pkg/report/cyclonedx/cyclonedx.go

```diff
@@ -31,12 +32,22 @@ func (w Writer) Write(report types.Report) error {
 	if err != nil {
 		return xerrors.Errorf("CycloneDX marshal error: %w", err)
 	}
+	return w.encodeBom(bom)
+}
 
+func (w Writer) WriteCoreComponents(component *core.Component) error {
```

Comment:
I aim to keep `pkg/report` and `pkg/k8s/report` independent. Can we add a tiny writer in `pkg/k8s/report/cyclonedx.go`?

```
// Writer implements types.Writer
type Writer struct {
	encoder cdx.BOMEncoder
	marshaler *cyclonedx.Marshaler
}

func NewWriter(output io.Writer, appVersion string) Writer {
	encoder := cdx.NewBOMEncoder(w.output, w.format)
	encoder.SetPretty(true)
	return Writer {
		encoder: encoder,
		marshaler: core.NewCycloneDX(appVersion),
	}
}

func (w Writer) Write(component *core.Component) error {
	bom, err := w.marshaler.Marshal(report)
	if err != nil {
		return xerrors.Errorf("CycloneDX marshal error: %w", err)
	}
	return w.encoder.Encode(bom)
}
```